tests/test_sourcefinder/test_image.py

```diff
@@ -402,3 +403,16 @@ def testMaskedBackgroundBlind(self):
             accessors.open(fits_file), radius=1.0)
         result = self.image.extract(det=10.0, anl=3.0)
         self.assertFalse(result)
+
+class TestFailureModes(unittest.TestCase):
+    """
+    If we get pathological data we should probably throw an exception
+    and let the calling code decide what to do.
+    """
+    def testFlatImage(self):
+        sfimage = accessors.sourcefinder_image_from_accessor(
+            SyntheticImage(data=np.zeros((512,512))))
+        self.assertTrue(np.ma.max(sfimage.data) == np.ma.min(sfimage.data),
+                        msg = "Data should be flat")
+        with self.assertRaises(RuntimeError):
```

Comment:
Isn't it an idea to come up with some of our own exceptions? It is very inlikely but there can also be some other piece of code that could raise a RuntimeError.

Probably completely out of scope for this PR.